### ISSUE TITLE
Pre-Trigger causes 409 on second create execution

### ISSUE THREAD

**Issue by wahyuen** [Azure/azure-cosmos-dotnet-v3#667]
>We are continuously addressing and improving the SDK, if possible, make sure the problem persist in the [latest SDK version](https://www.nuget.org/packages/Microsoft.Azure.Cosmos).

**Describe the bug**
We have an existing pretrigger which we have created using the v2 of the SDK, essentially a trigger that acts as a sequence generator by counting the number of documents in a query, then increment a property for the new document being created.

This appears to work fine when the trigger is created using the v2 SDK, however, this seems to throw a 409 with no reason when the trigger is created using the v3 SDK.

**To Reproduce**
Trigger code remains the same in both versions of the SDK

```csharp
            var trigger = new TriggerProperties
            {
                Id = SetJobNumberTrigger,
                TriggerType = TriggerType.Pre,
                TriggerOperation = TriggerOperation.Create,
                Body = @"function setJobNumber() {
                    var context = getContext();
                    var request = context.getRequest();      
                    var containerManager = context.getCollection();
                    var containerLink = containerManager.getSelfLink()

                    var documentToCreate = request.getBody();

                    var jobNumberQuery = ""SELECT VALUE MAX(r.jobNumber) from root r WHERE r.investigationKey = '"" + documentToCreate['investigationKey'] + ""'"";
                    containerManager.queryDocuments(containerLink,
                        jobNumberQuery,
                        function(err, countValue) {
                            if (err) throw new Error(err.message);
                            documentToCreate['jobNumber'] = (countValue.length > 0 ? countValue[0] : 0) + 1;
                        });

                    // update the document that will be created
                    request.setBody(documentToCreate);
                    }",
            };

        await container.Scripts.CreateTriggerAsync(trigger);
```

Unit test to illustrate expected behaviour when creating a document
```csharp
        [Fact]
        public async Task Jobs_ReturnsObject()
        {
            var value = new Job {Id = Guid.NewGuid(), InvestigationKey = "investigation~1"};

            // this should create the document successfully with jobnumber of 1
            await _fixture.LocalCosmosClient.CreateDocumentAsync(value, new ItemRequestOptions
            {
                PreTriggers = new List<string> { "SetJobNumber" }
            });

            var result = _fixture.LocalCosmosClient.Jobs().Where(x => x.InvestigationKey == "investigation~1").ToList();
            
            result.ShouldNotBeNull();
            result.OrderBy(x => x.JobNumber).Last().Id.ShouldBe(value.Id);

            result.OrderBy(x => x.JobNumber).Last().JobNumber.ShouldBe(result.Count);

            value.Id = Guid.NewGuid();

            // this should create the document successfully with jobnumber of 2
            await _fixture.LocalCosmosClient.CreateDocumentAsync(value, new ItemRequestOptions
            {
                PreTriggers = new List<string> { "SetJobNumber" }
            });

            result = _fixture.LocalCosmosClient.Jobs().Where(x => x.InvestigationKey == "investigation~1").ToList();

            result.ShouldNotBeNull();
            result.Count.ShouldBe(2);
            result.OrderBy(x => x.JobNumber).Last().Id.ShouldBe(value.Id);

            result.OrderBy(x => x.JobNumber).Last().JobNumber.ShouldBe(result.Count);
        }
```

**Expected behavior**
Documents should successfully be created and the trigger successfully run.

**Actual behavior**
After the first document has been created, the 2nd document that is created results in a 409 error.

**Environment summary**
SDK Version: 3.1.0
OS Version (e.g. Windows, Linux, MacOSX) Windows

**Additional context**
Add any other context about the problem here (for example, complete stack traces or logs).

```csharp
Microsoft.Azure.Cosmos.CosmosException : Response status code does not indicate success: 409 Substatus: 0 Reason: ().
   at Microsoft.Azure.Cosmos.ResponseMessage.EnsureSuccessStatusCode()
   at Microsoft.Azure.Cosmos.CosmosResponseFactory.ToObjectInternal[T](ResponseMessage cosmosResponseMessage, CosmosSerializer jsonSerializer)
   at Microsoft.Azure.Cosmos.CosmosResponseFactory.<CreateItemResponseAsync>b__4_0[T](ResponseMessage cosmosResponseMessage)
   at Microsoft.Azure.Cosmos.CosmosResponseFactory.ProcessMessageAsync[T](Task`1 cosmosResponseTask, Func`2 createResponse)
   at Soze.Common.Extensions.CosmosClientExtensions.CreateDocumentAsync[T](CosmosClient cosmosClient, T objectToSave, ItemRequestOptions options) in F:\MSWorkspace\soze.common\Soze.Common\Extensions\CosmosClientExtensions.cs:line 191
   at Soze.Common.Tests.Extensions.CosmosClientExtensionsTests.Jobs_ReturnsObject() in F:\MSWorkspace\soze.common\Soze.Common.Tests\Extensions\CosmosClientExtensionsTests.cs:line 310
--- End of stack trace from previous location where exception was thrown ---
```

**Comment by j82w** [comment]
@wahyuen can you take a look at this [PR](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/668)? I tried to repo the issue, but my test passed. Are you using any custom serialization setting for camel case?

**Comment by wahyuen** [comment]
@j82w Apologies, upon looking at your PR and recreating it, then analysing the diff between that and our own code, I managed to get to the root cause of the issue being that we incorrectly ported our v2 code in terms of how we define unique keys upon container construction.

We had a composite key on that container, which in v2 we had a single `UniqueKey` with multiple paths on it. When I ported to v3, this was inadvertantly changed to multiple `UniqueKey` and a single path on each.

Completely my own fault and once that was corrected, the test conditions seem to work fine!

Only feedback I would add is perhaps if there was a way for the 409 response to return the underlying reason of why the trigger failed? (unsure if we need to code this into the trigger or if we might get this natively?) The lack of the reason left us stumped in terms of what to look for in diagnosing the issue.

Thanks again for taking the time to create the PR!
